tools/port-drayage-webservice/src/main/java/com/leidos/inspection/InspectionActions.java

```diff
@@ -0,0 +1,194 @@
+
+            // If no current action set as current action
+            if (currentInspection == null) {
+                currentInspection = inspectionStatus;
+            }
+
+            // else add to list of pending action
+            else {
+                pendingInspection.addInspectionsItem(inspectionStatus);
+            }
+        } else {
+            logger.warn("Attempted to add null InspectionRequest!");
+        }
+
+    }
+
+    /**
+     * Method to get most recent {@link InspectionStatus} for a given vehicle id.
+     * First searches pending inspections, then current inspection, then completed
+     * inspections and returns the first match.
+     * 
+     * @param vehicleId {@link String} valid non null vehicleID
+     * @return {@link InspectionStatus} most recent inspection for a given vehicle
+     *         id
+     */
+    public InspectionStatus getInspectionStatus(String vehicleId) {
+        if (vehicleId != null) {
+            // Pending actions ( null check since arraylist is initially null )
+            if (pendingInspection.getInspections() != null) {
+                for (InspectionStatus inspectionStatus : pendingInspection.getInspections()) {
+                    if (vehicleId.equals(inspectionStatus.getVehicleId())) {
+                        return inspectionStatus;
+                    }
+                }
+            }
+            // Current action
+            if (currentInspection != null && vehicleId.equals(currentInspection.getVehicleId())) {
+                return currentInspection;
+            }
+            // Completed actions ( null check since arraylist is initially null )
+            if (completedInspections.getInspections() != null) {
+                for (InspectionStatus inspectionStatus : completedInspections.getInspections()) {
+                    if (vehicleId.equals(inspectionStatus.getVehicleId())) {
+                        return inspectionStatus;
+                    }
+                }
+            }
+            logger.warn(String.format("No current inspection action for vehicle ID %s !", vehicleId));
+            return null;
+        }
+        logger.warn("Null vehicle id is not valid!");
+        return null;
+    }
+
+    /**
+     * To complete inspection set {@link InspectionActions#currentInspection} status
+     * to {@link StatusEnum#PASSED}. Current inspection is then added to the
+     * completed inspections list and the first pending action , if any are present
+     * is set as the new current action. If no pending actions are present the
+     * current action is set to null.
+     * 
+     */
+    public void completeInspection() {
+        if (currentInspection != null) {
+            // Set current inspection to PASSED
+            currentInspection.setStatus(StatusEnum.PASSED);
+            currentInspection.setCompleted(System.currentTimeMillis());
+            completedInspections.addInspectionsItem(currentInspection);
```

Comment:
This is not a requirement. Currently the service tracks all completed actions as well as there start and stop times. I think this is useful for our testing since it can give us information about how much of our testing time was taken up by these actions.